e2e-tests/tfm/cypress/e2e/journeys/case-amendments/facility-end-date/feature-flag-disabled/gef-deal-expect-no-facility-end-date-features.spec.js

```diff
@@ -0,0 +1,85 @@
+import relative from '../../../../relativeURL';
+import facilityPage from '../../../../pages/facilityPage';
+import { ADMIN, BANK1_MAKER1, PIM_USER_1, T1_USER_1 } from '../../../../../../../e2e-fixtures';
+import { MOCK_APPLICATION_AIN } from '../../../../../fixtures/mock-gef-deals';
+import { MOCK_FACILITY_ONE } from '../../../../../fixtures/mock-gef-facilities';
+import { DEAL_TYPE } from '../../../../../../../gef/cypress/fixtures/constants';
+import amendmentsPage from '../../../../pages/amendments/amendmentsPage';
+import dateConstants from '../../../../../../../e2e-fixtures/dateConstants';
+
+if (Cypress.env('FF_TFM_FACILITY_END_DATE_ENABLED') === 'false') {
```

Comment:
this line should not be in the test - values should be picked up from test.yml

Reply:
The values are picked up from test.yml, however we've made it so it looks through the test files under `amendments` twice - once with `FF_TFM_FACILITY_END_DATE_ENABLED` as `true` and once with `FF_TFM_FACILITY_END_DATE_ENABLED` as `false`. This test should only be run when `FF_TFM_FACILITY_END_DATE_ENABLED` is `false`. So I'm happy with this implementation.

Reply:
@abhi-markan all instances of this still need to be removed as will be handled by test.yml and should only be handled from there

Reply:
We're currently looking at improving how we handle feature flags and the testing around them. We're looking to establish patterns to allow us to ensure that adding a feature flag does not cause regression issues.

We're doing this pattern (which does not look benificial here, but is in files in Marks comments [and we're doing it here to remain consistent]) to allow us to test both FF on and FF off. We run these tests twice. Once with the FF on, and once with FF off, seen in the `test.yml` as `  # 6.1 E2E - TFM backwards compatibility`. We do this to avoid regression issues with FFs.

Technically this test _could_ be its own file which is handled in test.yml *but* we want to run all tests with both TFM backwards compatiblity and with the ff enabled. If we were to do this through purely the `test.yml` we would have to duplicate many tests and only change a few lines of code (as mentioned, we're doing this with `unissued-facilities-change-all-AIN.spec.js` for a env var relating to this facility end date work -- and that file is probably more obvious why we are doing this here) -- we want to keep our tests DRY and consistent with ff on and off.

I'm happy to have a longer talk about this -- we're continously looking to improve how we do this. This approach has been done as a result of long discussions and an evolving methodology in regards to introducing feature flags to DTFS -- and we're continously reviewing this to make it better.

Also very happy to loop you in on this & invite you to meetings when we review this -- we have a ticket to remove this feature flag which will see us remove these tests. Then for the next feature flag we impliment we will either look to refine this pattern or try something new (our ultimate goal is to find the smoothest and least painful way of ensuring no regression failures when using a FF).

Reply:
Sorry that's a long comment. The tldr of it is: We're currently trying ways of handling feature flags and testing to avoid bugs in regression when the FF is off (as well as on). This is the way we've chosen for this FF and all facility end date tickets. This FF will be removed, and there is an existing ticket to remove it and these tests. We'll have a discussion about how we can improve this approach, and I'm happy to loop you in :)

Reply:
closing this comment shortly as i think this is something to pick up in dev friday/offline

Reply:
@abhi-markan

Reply:
Apologies everyone for a delay reply, we should not have a conditional branching in a Cypress test to test whether FF is on or off, these env variables will be picked up from `test.yml` (which I have refactored in #3390 PR).

Where a user journey (E2E) will be executed twice one with FF off or non-existent and one where it exist, in order to test such journeys where FF is on then those tests should be added to `test.yml` FF specific jobs matrix array.

Reply:
can all instances of this be removed in this PR

Reply:
We should have a meeting about this before removal -- I'll get my head around this again. Will send an invite out shortly

Reply:
Raised as another ticket -- will be covered there to keep this PR in scope